### PR TITLE
Datastore - Persistent Storage 

### PR DESCRIPTION
**Pull request by anushadatta**
Added Datastore functionality (Google App Engine API for Java) for persistent storage of blog comments. Created two servlets, LoadCommentsServlet and AddCommentServlet, to load all comments on the web page and add a new comment to the Datastore respectively.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java [hunk 1]
```diff
@@ -23,43 +29,47 @@
...
-    String text = request.getParameter("comment-text");
+    // Get BlogComment entities from Datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Query query = new Query("BlogComment").addSort("postDate", sortingOrder);
+    PreparedQuery results = datastore.prepare(query);
 
-    // Upload comment 
-    uploadComment(date, text);
+    // Create ArrayList<BlogComment> from entities
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:54]: Seems the comment is not needed here as it is obvious from code.

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java [hunk 2]
```diff
@@ -23,43 +29,47 @@
...
 
-      BlogComment c = new BlogComment(date, text);
+      BlogComment c = new BlogComment(postDate, text);
       commentsList.add(c);
+    }
+    
+    // Convert the comments data to JSON
+    String json = convertToJsonUsingGson(commentsList);
+
+    // Send the JSON as the response
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:69]: Consider defining this as constant. `private static final String JSON_CONTENT_TYPE = "application/json"`

**Review by rashikaranpuria — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java (hunk 1 above)
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:54]: Got you 😄

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java (hunk 2 above)
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:69]: It did not strike me to do that, that's a great suggestion. I'll definitely make that amendment!

**Review by rashikaranpuria — COMMENTED** [review]
LGTM

**Pull request merged**